assets/content/cookbook/Macros/compare-anon-structures.md

```diff
@@ -0,0 +1,131 @@
+# Compare Anonymous Structures and Objects
+
+## Why use a macro
+
+Sometimes it can be useful to compare two anonymous structures or objects
+ by their properties, even if they're not the same.
+
+For example, if we have two objects, `a` and `b`:
+
+```haxe
+var a = {
+  valA : 1,
+  valB : 2,
+  valC : 3,
+  valD : 4,
+};
+
+var b = {
+  valA : 1,
+  valB : 2,
+  valC : 3,
+  valD : 4,
+  someOtherValue : 'value'
+};
+```
+
+And wish to ensure `b`'s properties match all of `a`'s properties,
+ we would need the following construct:
+
+```haxe
+// Ensure b has all fields of a.
+
+trace( a.valA == b.valA &&
+       a.valB == b.valB &&
+       a.valC == b.valC &&
+       a.valD == b.valD );
+```
+
+Since a simple `==` comparison yields an error; the objects are
+ not of the same type.
+We can automatize this comparison using Macros.
+
+## Macro
+
+```haxe
+import haxe.macro.Context;
+import haxe.macro.Expr;
+import haxe.macro.Type;
+
+class CompareMacro {
+  macro static public function matchAllFromAnon(reference:Expr, shouldHave:Expr):Expr {
+    // The expressions that must be verified to assure equality
+    //  i.e., individual ref.val == shouldHave.val assertions
+    var conds:Array<Expr> = [];
+    
+    var curPos = Context.currentPos();
+    
+    // The fields of the reference object
+    var fieldsToCheck = [];
+    
+    var typeOfReference = Context.typeof(reference);
+    switch(typeOfReference) {
```

Comment:
This entire part should be in a loop which is called recursively in the `TType` case.